src/test_dll.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python
+# -*- coding: utf -8 -*-
+import pytest
+
+
+from dll import Node, DoublyLL
+
+
+INIT_TABLE = [
+ (1, 1, 1, 1),
+ ([1, 2, 3], 3, 1, 3)
+]
+
+LEN_TABLE = [
+    (None, 0),
+    (1, 1),
+    ([1, 2, 3], 3)
+]
+
+APPEND_TABLE = [
+    (None, 1, 1),
+    (1, 2, 2),
+    ([1, 2, 3], 5, 4)
+]
+
+POP_TABLE = [
+    (1, 1, 0),
+    ([1, 2, 3], 3, 2)
+]
+
+SHIFT_TABLE = [
+    (1, 1, 0),
+    ([1, 2, 3], 1, 2)
+]
+
+REMOVE_TABLE = [
+    (1, 5, False, 1),
+    (1, 1, True, 0),
+    ([1, 2, 3], 5, False, 3),
+    ([1, 2, 3], 1, True, 2),
+    ([1, 2, 3], 3, True, 2),
+    ([1, 2, 3], 2, True, 2),
+    ([1, 2, 3, 4, 5, 6, 7], 5, True, 6)
+]
+
+
+def test_init_Node():
+    node1 = Node(5)
+    assert node1.value == 5
+    assert not node1.previous_node
```

Comment:
You should only have one assert statement per test, so you will be more clear (to yourself) about what exactly is failing. If there are multiple things that need to be checked, put them in separate tests. If its getting tedious typing out lots of tests, its time to start thinking about how to save some effort using fixtures. I can help you with this if you want.